docs/docs/getting-started/developing-on-hydra.md

```diff
@@ -0,0 +1,89 @@
+This is a synchronous process and requires all the head participants to be online all the time, the tradeoff is that there is no uncertainty or possibility of rollbacks, the moment the snapshot is confirmed there is no going back.
+
+For the Pub side of the API, an important endpoint will be `NewTx`. This is used to submit a transaction to the node. There are a couple of ways the transaction can be submitted, but the simplest method we found is to simply serialise the transaction built with `Cardano.Api` to CBOR and submit that through a websocket.
+
+## Testing
+
+Testing inside the hydra head can be quite different than for normal DApps, the reason is you will not have access to EmulatorTrace or an easy way to spin up a dev environment to test.
+
+However it is possible to spin up a cardano-node, and a cluster of hydra-nodes programmatically and run tests directly on that.
+The module hydra-cluster provides functions to spin up the set of required nodes and to run assertions on their output.
+
+A couple of steps are required to set up the initial environment:
+
+- We need to spin up at least one Cardano BFT node. This can be done with `withBFTNode` which takes the cardano node configuration and a callback that has access to an instance of the running node. The node parameters can be generated by a helper also exported by hydra-cluster: `newNodeConfig`.
+
+- In the callback, we can then spin up a cluster of hydra-nodes using `withHydraCluster`. This will require the cardano-node socket, which can be obtained from the running node instance, and a set of cardano and hydra keys which can also be easily generated through helpers exposed by `hydra-cluster`.
+
+Let's walk through a slightly simplified version of one of our tests, with comments explaining the setup we just described.
+This test will spin up a single cardano node, and a pair of hydra nodes (Alice and Bob).
+
+```haskell
+-- Tracer is an instance of https://hackage.haskell.org/package/contra-tracer and is used by several
+-- hydra components to deal with logs in a structured way
+it "spins up two hydra nodes" $ \tracer ->
+  -- Set a timeout for the whole test to fail
+  failAfter 60 $
+    -- We create a temporary directory to contain all the files required to spin up a node
+    withTempDir "cardano-node-tmp-dir" $ \cTmpDir -> do
+      config <- newNodeConfig cTmpDir
+      -- This starts the BFTNode as described above
+      withBFTNode (contramap FromCardanoNode tracer) config $ \(RunningNode _ nodeSocket) -> do
+        -- We create a temporary directory to contain all the files required to spin up the hydra nodes
+        withTempDir "hydra-nodes-tmp-dir" $ \hTmpDir -> do
```

Comment:
I would recommend to re-use the temp dir to make this simpler. No point in separating, both will now (soon #430) write to `logs/cardano-node.log` and `logs/hydra-node-x.log`.